netbox_netprod_importer/vendors/cisco/nxos.py

```diff
@@ -125,3 +120,43 @@ def get_detailed_lldp_neighbours(self):
                 "port": n["port_id"],
                 "chassis_id": n["chassis_id"]
             }
+
+    def get_detailed_cdp_neighbours(self):
+        """
+        Napalm does not support cdp
+
+        :return neighbours: [{
+                "local_port": local port name,
+                "hostname": neighbour hostname (if handled),
+                "port": neighbour port name,
+            }]
+        """
+        cmd = "show cdp neighbors detail" + self._driver_end_selection()
+
+        cmd_output = self.device.cli([cmd])[cmd]
+        neighbours = self._correct_and_convert_to_dict(cmd_output)["TABLE_cdp_neighbor_detail_info"][
+            "ROW_cdp_neighbor_detail_info"]
+
+        if isinstance(neighbours, dict):
+            neighbours = [neighbours]
+
+        for n in neighbours:
+            yield {
+                "local_port": n["intf_id"],
+                "hostname": re.sub("\(.*$", "", n["device_id"], count=1),
+                "port": n["port_id"],
+            }
+
+    def _driver_end_selection(self) -> str:
```

Comment:
I don't really get these changes. I will try to test it on a Nexus 9000 tomorrow.
Could you explain why is there a difference between the nxapi and nxos_ssh here please?

Reply:
Response
{
  "jsonrpc": "2.0",
  "error": {
    "code": -32602,
    "message": "Invalid params",
    "data": {
      "msg": "Request contains invalid special characters"
    }
  },
  "id": 1
}

Reply:
Software
  BIOS: version 07.59
  NXOS: version 7.0(3)I4(8a)
  BIOS compile time:  08/26/2016
  NXOS image file is: bootflash:///nxos.7.0.3.I4.8a.bin
  NXOS compile time:  7/5/2018 11:00:00 [07/06/2018 02:57:12]